storage_service/storage_service/settings/base.py

```diff
@@ -181,6 +189,39 @@ def get_env_variable(var_name):
 # See: https://docs.djangoproject.com/en/dev/ref/settings/#authentication-backends
 
 AUTHENTICATION_BACKENDS = ["django.contrib.auth.backends.ModelBackend"]
+
+try:
+    PASSWORD_MINIMUM_LENGTH = int(environ.get("SS_AUTH_PASSWORD_MINIMUM_LENGTH", 8))
+except ValueError:
+    PASSWORD_MINIMUM_LENGTH = 8
+
+AUTH_PASSWORD_VALIDATORS = [
+    {
+        "NAME": "django.contrib.auth.password_validation.MinimumLengthValidator",
+        "OPTIONS": {"min_length": PASSWORD_MINIMUM_LENGTH},
+    }
+]
+
+# Additional password validation is enabled by default. To disable, set the
+# following AUTH_PASSWORD_* settings to false in the environment.
+if not is_false(environ.get("SS_AUTH_PASSWORD_COMMON_VALIDATION", "")):
+    AUTH_PASSWORD_VALIDATORS += [
+        {"NAME": "django.contrib.auth.password_validation.CommonPasswordValidator"}
+    ]
+
+if not is_false(
+    environ.get("SS_AUTH_PASSWORD_USER_ATTRIBUTE_SIMILARITY_VALIDATION", "")
+):
+    AUTH_PASSWORD_VALIDATORS += [
+        {
+            "NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator"
+        }
+    ]
+
+if not is_false(environ.get("SS_AUTH_PASSWORD_COMPLEXITY_VALIDATION", "")):
+    AUTH_PASSWORD_VALIDATORS += [
+        {"NAME": "administration.validators.PasswordComplexityValidator"}
+    ]
```

Comment:
Yeah, agreed, the current `base.py` settings file here is not great. I'll look at refactoring this into modules, more akin to what's being done in the Dashboard, as at a quick glance I don't think that would take very much time or confer much risk. I also like the approach you're suggesting here so I'll keep an eye toward either mixing the two, or deferring to something like your diff if a proper refactor seems like more risk/time once I get into it.

Reply:
In commit https://github.com/artefactual/archivematica-storage-service/pull/553/commits/02327ed6a395fb95975795745c57b0426b975cca, I've moved the auth settings other than `AUTHENTICATION_BACKENDS` into a new auth components module, similar to how the LDAP/Shibboleth/CAS settings are organized in Archivematica proper. I opted for this instead of something like the suggested `get_my_auth_settings()` function (despite seeing some advantages there, such as for testing), as I think the existing Archivematica will extend better to other components such as LDAP/Shibboleth/CAS, which set a number of settings variables that won't necessary be easily returned as a list, for example. Does that seem reasonable to you?

After looking at it a bit, I'd rather not touch the other settings quite yet, as I feel like there's the potential for accidental regressions and the time required to set up environments to test and make sure everything is working correctly seems significant enough to be out-of-scope for this project. That said, hopefully you agree that what I have done here suffices as a first step in the direction of more modularity.